examples/study/study_guide.py

```diff
@@ -0,0 +1,84 @@
+from agentic.tools.rag_tool import RAGTool
+
+from agentic.common import Agent, AgentRunner
+from agentic.models import GPT_4O_MINI
+# This is the "hello world" agent example. A simple agent with a tool for getting weather reports.
```

Comment:
Suggest you update the comment with a fuller description of this agent and how it works. Note your PR comment will be lost to history, but this comment in code will live forever.

The comment should explain how to setup and run the agent. Also , does the RAG indexing happen every time or only once?